### ISSUE TITLE
Track User Story 14: BG 903 misplaced: raises linkedBGMissed error

### ISSUE THREAD

**Issue by UweSteinkeFromSiemens** [openETCS/modeling#971]
BG 903 is reported as missed. When this error flag is raised, the observations are:

- BG 903 is not marked as passed so far.
- BG 903: expected location is 
  - 9491 +- 244 = 9241  <->  9735
- current train position: 
  - 9882 -103/+104 = 9779 <-> 9986
- extended train position (comprising sampling uncertainty): 
  - 9882 -132/+104 = 9750 <-> 9986

Due to 9735 < 9750, the train left BG903s expectation window without detecting it on the track. 

The location of BG 903 has to be corrected.

**Comment by UweSteinkeFromSiemens** [comment]
An alternative eventuality for the observed symptom may be caused, if the test environment / test track simulation is only loosely coupled with the EVC and the BG detection and the odometry is arbitrary slipping around. 

@jokaICS : Does the test environment with simctrl and TCP/IP preserve the clock synchronism between the test environment and the EVC model?

**Comment by jokaICS** [comment]
The test track runs within the EVC sim, so there is no delay for triggered balise telegrams/ radio messages due to TCP/IP (there is of course a small delay in the visualization of radio/balise messages on the SimCtrl UI).

A question regarding linking information (P005): the distance for the *first* announced BG is the distance from its predecessor (and not the distance from the current train position), correct? Probably a stupid question, but when I started designing the test track it was not obvious from the SRS (for me, at least).

**Comment by UweSteinkeFromSiemens** [comment]
The distance for the first announced BG refers to the just overpassed BG (if announced via BG) or to the LRBG (if announced via radio).

**Comment by jokaICS** [comment]
OK, but then the announcement of BG903 in the "Test Track" test.trk should be correct:
![image](https://cloud.githubusercontent.com/assets/13429700/11652932/e1a70656-9d9a-11e5-9be9-33f1623e7497.png)

15m (position of BG902) + 25m (d_link(0)) is 40m (position of BG903).

**Comment by UweSteinkeFromSiemens** [comment]
Further analysis of another test run, when the linkedBGMissed is raised : 

1 clock cycle before the error flag is raised: 
- The current train position is within the expectation window of BG 903. 

In the same cycle, when the error flag is raised: 
- BG 903 is just reported as overpassed and therefore the locations of all BGs and the current train position are recalculated, reducing the location uncertainties of the BG 903. 

- The reduced uncertainties shrink the expectation window so that the current train position drops outside of it. 

- The information, that the BG has been passed in the same cycle, is ignored, caused by internal housekeeping of the BGs list. 

In summary, the combination of overpassing, shrinking out of the expectation window and housekeeping in the same cycle causes the effect. 

The observations need to be verified deeply before the bug can be fixed.

**Comment by UweSteinkeFromSiemens** [comment]
Fix provided with commit 13a58f391a1a610e4c14565b165215fe1cb907ef.

**Comment by UweSteinkeFromSiemens** [comment]
If BG 903 is not being missed any more, this issue can be closed.
@BerndHekele, @Farhangi : feedback would be useful.

**Comment by BerndHekele** [comment]
Tested succeffully
